osutil/flock_test.go

```diff
@@ -89,6 +113,50 @@ func (s *flockSuite) TestLockUnlockWorks(c *C) {
 	c.Assert(cmd.Run(), IsNil)
 }
 
+// Test that ReadLock and Unlock work as expected.
+func (s *flockSuite) TestReadLockUnlockWorks(c *C) {
+	if os.Getenv("TRAVIS_BUILD_NUMBER") != "" {
+		c.Skip("Cannot use this under travis")
+		return
+	}
+	if !flockSupportsConflictExitCodeSwitch(c) {
```

Comment:
not for this PR, but could we also test this not by using the `flock` command, but by using the functions we have implemented in go routines? Is this considered too unstable or difficult to write?

Reply:
The problem of testing it with goroutines is that we could easily be testing a broken implementation with a broken test helper. While flock is probably super easy in practice and our code is correct, in general, it might be wise to test using a separate, well-established implementation.

Having said that, for flock, I would definitely see this as an advantage. I may do this as a follow-up but if you want to jump right at it :)